dvc/dependency/repo.py

```diff
@@ -46,14 +47,33 @@ def _make_repo(self, **overrides):
         with external_repo(**merge(self.def_repo, overrides)) as repo:
             yield repo
 
-    def status(self):
-        with self._make_repo() as repo:
-            current = repo.find_out_by_relpath(self.def_path).info
+    def _get_checksum(self, updated=False):
+        rev_lock = None
+        if not updated:
+            rev_lock = self.def_repo.get(self.PARAM_REV_LOCK)
 
-        with self._make_repo(rev_lock=None) as repo:
-            updated = repo.find_out_by_relpath(self.def_path).info
+        try:
+            with self._make_repo(rev_lock=rev_lock) as repo:
+                try:
+                    return repo.find_out_by_relpath(self.def_path).info["md5"]
+                except OutputNotFoundError:
+                    # Fall through to after exception handler
+                    path = os.path.join(repo.root_dir, self.def_path)
```

Comment:
You have the same line in #68 ,why duplicating it?

Reply:
Oh, I see you have a second except logic here.

Reply:
I'm wondering if it's good to use `repo` outside of the `_make_repo`.

Reply:
@fabiosantoscode But that is not true, you are using `cached_clone` down below, so you are using it and not the `_make_repo`s part.

Reply:
Though I can see the issue here, that is a gotcha for me 🙂